src/vs/workbench/api/electron-browser/mainThreadTreeViews.ts

```diff
@@ -158,3 +189,100 @@ export class TreeViewDataProvider implements ITreeViewDataProvider {
 		}
 	}
 }
+
+// {{SQL CARBON EDIT}}
+export class OETreeViewDataProvider implements sqlops.ObjectExplorerProvider {
```

Comment:
seems like this class should be located in the `sql` folder somewhere since it's a bit more than a target patch.  Perhaps even some of the other large code block patches could be written as functions under `sql` that are called from there.  I think that's more inline with the general approach we've taken to updating vscode sources.

Reply:
I actually can't move this out without moving a lot more things out or causing a circular dependency.